src/main/java/com/segment/analytics/android/integrations/intercom/IntercomIntegration.java

```diff
@@ -43,29 +50,211 @@ public String key() {
     String appId = settings.getString("appId");
 
-    Intercom.initialize(application, apiKey, appId);
+    Intercom.initialize(application, mobileApiKey, appId);
+    this.intercom = provider.get();
     this.logger = logger;
   }
 
   @Override
   public void identify(IdentifyPayload identify) {
     super.identify(identify);
+
+    String userId = identify.userId();
+
+    if (isNullOrEmpty(userId)) {
+      intercom.registerUnidentifiedUser();
+      logger.verbose("Intercom.client().registerUnidentifiedUser()");
+    } else {
+      Registration registration = Registration.create().withUserId(userId);
+      intercom.registerIdentifiedUser(registration);
+      logger.verbose("Intercom.client().registerIdentifiedUser(registration)");
+    }
+
+    Map<String, Object> intercomOptions = identify.integrations().getValueMap("Intercom");
+    if (!isNullOrEmpty(intercomOptions)
+        && !isNullOrEmpty(String.valueOf(intercomOptions.get("userHash")))) {
+      intercom.setUserHash(String.valueOf(intercomOptions.get("userHash")));
+    }
+
+    Traits traits = identify.traits();
+    if (!isNullOrEmpty(traits) && (!isNullOrEmpty(intercomOptions))) {
+      setUserAttributes(traits, intercomOptions);
+      return;
+    }
+    setUserAttributes(traits, null);
   }
 
   @Override
   public void track(TrackPayload track) {
     super.track(track);
+
+    String eventName = track.event();
+    Properties properties = new Properties();
+    properties.putAll(track.properties());
+    Map<String, Object> propertiesMap = new HashMap<>();
+
+    if (!isNullOrEmpty(properties)) {
+      for (Map.Entry<String, Object> entry : properties.entrySet()) {
+        String trait = entry.getKey();
+        Object value = entry.getValue();
+        propertiesMap.put(trait, value);
+      }
+    }
+    intercom.logEvent(eventName, propertiesMap);
+    logger.verbose("Intercom.client().logEvent(%s, %s)", eventName, propertiesMap);
   }
 
   @Override
   public void group(GroupPayload group) {
     super.group(group);
+
+    if (isNullOrEmpty(group.groupId())) return;
+
+    UserAttributes.Builder userAttributes = new UserAttributes.Builder();
+    Traits traits = new Traits();
+    traits.putAll(group.traits());
+    traits.put("id", group.groupId());
+    Company company = setCompany(traits);
+    userAttributes.withCompany(company);
+    intercom.updateUser(userAttributes.build());
+    logger.verbose("Intercom.client().updateUser(%s)", userAttributes);
+  }
+
+  public void reset() {
+    intercom.logout();
+    logger.verbose("Intercom.client().reset()");
+  }
+
+  private void setUserAttributes(Traits realTraits, @Nullable Map<String, Object> intercomOptions) {
+    Traits traitsCopy = new Traits();
+    traitsCopy.putAll(realTraits);
+    traitsCopy.remove("userId");
+    traitsCopy.remove("anonymousId");
+
+    String name = traitsCopy.name();
+    String email = traitsCopy.email();
+    String phone = traitsCopy.phone();
```

Comment:
What if phone is of type integer?

Reply:
Actually, not sure, but phone is specced as String data type, so we shouldn't allow users to send non-String phone values.